src/Databases/DDLDependencyVisitor.cpp

```diff
@@ -387,34 +387,32 @@ namespace
 
             const auto & arg = args[arg_idx];
 
-            if (evaluate)
+            if (const auto * id = arg->as<ASTIdentifier>())
+                return id->name();
+
+            if (const auto * literal = arg->as<ASTLiteral>())
```

Comment:
If the argument of a function is a string literal, and `evaluate = true`, we can return early and avoid copying the context.